### PR TITLE
[UI] Configurable Allocator for modm::ViewStack

### PR DESCRIPTION
**Pull request by MatthewMArnold**
Having a custom allocator allows users to control how `modm::AbstractView` elements stored by the view stack are allocated. This for example, allows the user to statically allocate all views and removes the memory and processing overhead of dynamically allocating views. For an example, see aruw-mcb's [`MainMenu` class](https://github.com/uw-advanced-robotics/aruw-mcb/blob/develop/aruw-mcb-project/src/aruwsrc/display/main_menu.cpp). By default, a dynamic allocator used, so default behavior remains unchanged.

This also has a couple minor tweaks to other UI related components.

This is a follow-on to https://github.com/modm-io/modm/pull/949 and https://github.com/salkinium/modm/tree/JoshuaT/ModmUpgrade2 that contains outstanding commits that were never applied cherry-picked on top of latest modm-io/modm develop.

### CODE REVIEW

**Comment by MatthewMArnold** [comment]
@OroArmor @calebchalmers here you go

File: src/modm/ui/menu/abstract_menu.hpp [hunk 1]
```diff
@@ -30,12 +30,17 @@ namespace modm{
 	 * \author Thorsten Lajewski
 	 * \ingroup modm_ui_menu
 	 */
-	class AbstractMenu: public AbstractView
+	template<typename Allocator = allocator::Dynamic<IAbstractView> >
```
Comment (salkinium) [inline src/modm/ui/menu/abstract_menu.hpp]: Could you check if we can instead use `std::allocator`? I think `modm::allocator` was just for AVRs, but now we have a libstdc++ for it.

Comment (salkinium) [reply src/modm/ui/menu/abstract_menu.hpp]: I'm not allowed to push changes to this branch:
```
 $ git push uw-advanced-robotics
ERROR: Permission to uw-advanced-robotics/modm.git denied to salkinium.
```

So please [cherry-pick this commit over to your branch](https://github.com/salkinium/modm/commit/3ccda11c0c227688963d17af1330bae0401da7d4) and let's see if the CI passes.

**Review by salkinium — APPROVED** [review]
Awesome! Could you rebase on develop again to apply the fixup?

**Comment by salkinium** [comment]
The fixup is still there. Try `git rebase -i develop` to apply the fixup to the right commit.

**Comment by MatthewMArnold** [comment]
> The fixup is still there. Try `git rebase -i develop` to apply the fixup to the right commit.

Ah I misunderstood what you were asking, should be done now

**Review by salkinium — APPROVED** [review]
Thanks!

**Pull request merged**